adapters/Pangle/PangleAdapter/Bidding/GADPangleRTBNativeRenderer.m

```diff
@@ -0,0 +1,199 @@
+    if (_nativeAd && _nativeAd.data) {
+        return _nativeAd.data.AdTitle;
+    }
+    return nil;
+}
+
+- (NSString *)body {
+    if (_nativeAd && _nativeAd.data) {
+        return _nativeAd.data.AdDescription;
+    }
+    return nil;
+}
+
+- (NSString *)callToAction {
+    if (_nativeAd && _nativeAd.data) {
+        return _nativeAd.data.buttonText;
+    }
+    return nil;
+}
+
+- (NSDecimalNumber *)starRating {
+    return [NSDecimalNumber decimalNumberWithString:[NSString stringWithFormat:@"%ld",(long)_nativeAd.data.score]];
+}
+
+- (NSString *)store {
+    return nil;
+}
+
+- (NSString *)price {
+    return nil;
+}
+
+- (NSString *)advertiser {
+    if (_nativeAd && _nativeAd.data) {
+        return _nativeAd.data.AdTitle;
+    }
+    return nil;
+}
+
+- (NSDictionary<NSString *,id> *)extraAssets {
+    if (_nativeAd) {
+        return @{BUDNativeAdTranslateKey:_nativeAd};
+    }
+    return nil;
+}
+
+#pragma mark BUNativeAdDelegate
+
+- (void)nativeAdDidLoad:(BUNativeAd *)nativeAd view:(UIView *_Nullable)view {
+    BUMaterialMeta *data = nativeAd.data;
+    // main image of the ad
+    if (data.imageAry && data.imageAry.count && data.imageAry[0].imageURL != nil){
+        _images = @[[self imageWithUrlString:data.imageAry[0].imageURL]];
+    }
+    
+    // icon image of the ad
+    if (data.icon && data.icon.imageURL != nil){
+        _icon = [self imageWithUrlString:_nativeAd.data.icon.imageURL];
+    }
+    
+    [_relatedView refreshData:nativeAd];
+    
+    if (_loadCompletionHandler) {
+        _delegate = _loadCompletionHandler(self,nil);
+    }
+}
+
+- (void)nativeAd:(BUNativeAd *)nativeAd didFailWithError:(NSError *_Nullable)error {
+    if (_loadCompletionHandler) {
+        _loadCompletionHandler(nil, error);
+    }
+}
+
+- (void)nativeAdDidBecomeVisible:(BUNativeAd *)nativeAd {
+    id<GADMediationNativeAdEventDelegate> delegate = _delegate;
+    [delegate reportImpression];
+}
+
+- (void)nativeAdDidClick:(BUNativeAd *)nativeAd withView:(UIView *_Nullable)view {
+    id<GADMediationNativeAdEventDelegate> delegate = _delegate;
+    [delegate reportClick];
+}
+
+- (void)nativeAd:(BUNativeAd *_Nullable)nativeAd adContainerViewDidRemoved:(UIView *)adContainerView {
+    
+}
+
+- (GADNativeAdImage *)imageWithUrlString:(NSString *)urlString {
+    NSURL *url = [NSURL URLWithString:urlString];
+    NSData *data = [NSData dataWithContentsOfURL:url];
+    UIImage *image = [UIImage imageWithData: data];
+    return [[GADNativeAdImage alloc] initWithImage:image];
+}
+
+
+- (void)didRenderInView:(nonnull UIView *)view
+       clickableAssetViews:
+           (nonnull NSDictionary<GADNativeAssetIdentifier, UIView *> *)clickableAssetViews
+    nonclickableAssetViews:
+        (nonnull NSDictionary<GADNativeAssetIdentifier, UIView *> *)nonclickableAssetViews
+         viewController:(nonnull UIViewController *)viewController {
+    [_nativeAd registerContainer:view withClickableViews:clickableAssetViews.allValues];
+}
+
```

Comment:
Can you confirm if [`-didUntrackView:`](https://developers.google.com/admob/ios/api/reference/Protocols/GADMediatedUnifiedNativeAd#/c:objc(pl)GADMediatedUnifiedNativeAd(im)didUntrackView:) isn't needed?

Reply:
it isn't needed.